.github/workflows/master.yml

```diff
@@ -37,18 +37,12 @@ jobs:
           echo "::add-matcher::.github/workflows/matchers/pylint.json"
           echo "::add-matcher::.github/workflows/matchers/black.json"
           echo "::add-matcher::.github/workflows/matchers/ruff.json"
-      - name: pre-cache
-        id: pre-cache
-        run: |
-          # Refresh cache daily
-          date=$(date -u "+%Y%m%d")
-          # Refresh cache if the action runner image has changed
-          image="$ImageOS-$ImageVersion"
-          echo "key=$image-$date" >> $GITHUB_OUTPUT
+      - uses: ./.github/actions/py-cache-key
+        id: py-cache-key
       - uses: actions/cache@v3
         with:
           path: .venv
-          key: ${{ steps.pre-cache.outputs.key }}-${{ steps.setup-python.outputs.python-version }}-${{ hashFiles('requirements/*requirements.txt') }}
+          key: ${{ steps.py-cache-key.outputs.key }}
```

Comment:
<img width="964" alt="image" src="https://github.com/mlflow/mlflow/assets/17039389/41523708-6a9e-4ea6-b402-5fc2c8cd2f09">

The generated key looks correct